packages/api-contract/src/base/types.ts

```diff
@@ -4,40 +4,58 @@
 import type { Observable } from 'rxjs';
 import type { SubmittableExtrinsic } from '@polkadot/api/submittable/types';
 import type { ApiTypes, ObsInnerType } from '@polkadot/api/types';
-import type { AccountId } from '@polkadot/types/interfaces';
+import type { AccountId, CodeSource } from '@polkadot/types/interfaces';
 import type { BN } from '@polkadot/util';
-import type { AbiMessage, BlueprintOptions, ContractCallOutcome, ContractOptions } from '../types';
+import type { AbiMessage, BlueprintOptions, ContractCallOutcome, ContractInstantiateOutcome, ContractOptions } from '../types';
 
-export interface BlueprintDeploy<ApiType extends ApiTypes> {
+export interface BlueprintQuery<ApiType extends ApiTypes> {
+  (origin: AccountId | string | Uint8Array, options: BlueprintOptions, ...params: unknown[]): ContractResult<ApiType, ContractInstantiateOutcome>;
+  (origin: AccountId | string | Uint8Array, value: bigint | BN | string | number, gasLimit: bigint | BN | string | number, source: string | Record<string, unknown> | CodeSource | Uint8Array, ...params: unknown[]): ContractResult<ApiType, ContractInstantiateOutcome>;
+  (origin: AccountId | string | Uint8Array, value: bigint | BN | string | number, gasLimit: bigint | BN | string | number, storageDepositLimit: bigint | BN | string | number, source: string | Record<string, unknown> | CodeSource | Uint8Array, ...params: unknown[]): ContractResult<ApiType, ContractInstantiateOutcome>;
+}
+
+export interface BlueprintTx<ApiType extends ApiTypes> {
   (options: BlueprintOptions, ...params: unknown[]): SubmittableExtrinsic<ApiType>;
   (value: bigint | string | number | BN, gasLimit: bigint | string | number | BN, ...params: unknown[]): SubmittableExtrinsic<ApiType>;
+  (value: bigint | string | number | BN, gasLimit: bigint | string | number | BN, storageDepositLimit: bigint | BN | string | number, ...params: unknown[]): SubmittableExtrinsic<ApiType>;
```

Comment:
We don't want these, i.e. we don't extend these into the future - it is purely for backwards compat. We expect everybody to use the options flavor. (Additionally this breaks compat completely, we don't like breaking user-space)

TL;DR This flavor _should_ have been removed in a previous major bump.

Reply:
So assuming this is for next major release, it's ok to remove both and similar ones?